kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/KieServicesClientImpl.java

```diff
@@ -123,6 +119,30 @@ private void init() {
 
     }
 
+    private List<String> getCapabilitiesFromServer() {
+        if (config.isJms()) {
+            if (config.isJmsTransactional()) {
+                kieServerInfo = getServerInfoInTransaction();
```

Comment:
@mswiderski, what I mean is this case. The client simply cannot get the capabilities from the server unless it has an access to a transaction manager. So one solution might be to add a reference to `UserTransaction` to the configuration and get server info in a transaction.